pkg/api/meta/meta.go

```diff
@@ -506,8 +548,20 @@ func (a genericAccessor) SetAnnotations(annotations map[string]string) {
 	*a.annotations = annotations
 }
 
-func (a genericAccessor) GetOwnerReferences() []unversioned.ObjectReference {
-	return a.ownerReferences
+func (a genericAccessor) GetOwnerReferences() []OwnerRefereceAccessor {
```

Comment:
+1 to not handing mutable array references out. Either copy and return or give index accessors